src/build_context.rs

```diff
@@ -54,27 +54,59 @@ impl BridgeModel {
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum ProjectLayout {
     /// A rust crate compiled into a shared library with only some glue python for cffi
-    PureRust,
+    PureRust(String),
     /// A python package that is extended by a native rust module.
     ///
     /// Contains the canonicialized (i.e. absolute) path to the python part of the project
-    Mixed(PathBuf),
+    Mixed {
+        python_module: PathBuf,
+        rust_module: PathBuf,
+        extension_name: String,
+    },
 }
 
 impl ProjectLayout {
     /// Checks whether a python module exists besides Cargo.toml with the right name
     pub fn determine(project_root: impl AsRef<Path>, module_name: &str) -> Result<ProjectLayout> {
-        let python_package_dir = project_root.as_ref().join(module_name);
-        if python_package_dir.is_dir() {
-            if !python_package_dir.join("__init__.py").is_file() {
+        let parts: Vec<&str> = module_name.split('.').collect();
+        let (python_module, rust_module, extension_name) = if parts.len() > 1 {
+            let mut rust_module = project_root.as_ref().to_path_buf();
+            rust_module.extend(&parts[0..parts.len() - 1]);
```

Comment:
Isn't `parts[0..parts.len() - 1]` just the entire slice?

Reply:
No, `parts[0..=parts.len() - 1]` is the entire slice. https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f48d54867e7a4879cf761036d71fab5d